bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -241,8 +241,10 @@ public boolean readCharacteristic(BluetoothCharacteristic characteristic) {
         scheduler.submit(() -> {
             try {
                 byte[] value = c.readValue();
-                characteristic.setValue(value);
-                notifyListeners(BluetoothEventType.CHARACTERISTIC_READ_COMPLETE, characteristic,
+                // create a copy of characteristic to make sure we don't interfere with writes
+                BluetoothCharacteristic copyChar = characteristic.copy();
```

Comment:
BluetoothCharacteristic could be even immutable, then it's clear that object can't be modified after creation. 

e.g.
BluetoothCharacteristic c = new BluetoothCharacteristic.Builder(characteristic).withValue(value).build();

Reply:
I don't think that the builder pattern adds much benefit in this case. There are a lot of methods in BluetoothCharacteristic that mutate the contained value so if you want it to be immutable might as well rip out the value entirely and go with the other PR.

Reply:
Maybe a better approach would be to make sure that BluetoothDevice makes a copy of all its characteristics in the getter methods to make the stored characteristics effectively immutable.

Reply:
Maybe remove the setter and use a `short` builder? Or make the second constructor public. 

```
final byte[] value;

public BluetoothCharacteristic(...) {
   this(..., new byte[0]);
}

private BluetoothCharacteristic(..., byte[] value) {
    ...
    this.value = value;
}

public BluetoothCharacteristic withValue(byte [] value) {
    return new BluetoothCharacteristic(..., value);
}
```

Reply:
@J-N-K Still doesn't solve the problem of what to do with all the value mutator methods in BluetoothCharacteristic. I wouldn't go with a mixed approach with my other PR by doing what you suggest above while moving the mutator methods in its own utility class. 
That said, none of the existing bindings or proposed binding even use the mutators so there should be no impact of moving them.